datafusion/expr/src/logical_plan/dml.rs

```diff
@@ -39,6 +39,8 @@ pub struct CopyTo {
     pub format_options: FormatOptions,
     /// SQL Options that can affect the formats
     pub options: HashMap<String, String>,
+    /// Hive Options that can affect hive-style partitioning
+    pub hive_options: HashMap<String, String>,
```

Comment:
I don't think we should separate out the hive options within the LogicalPlan CopyTo struct. The existing options hashmap can hold the hive options as well. Prefixing them with "hive." is enough in my opinion.